app/javascript/admin/controllers/config_controller.js

```diff
@@ -93,6 +93,40 @@ export default class ConfigController extends Controller {
     }
   }
 
+  displaySnackbar(message) {
+    return document.dispatchEvent(
+      new CustomEvent('snackbar:add', { detail: { message } }),
+    );
+  }
+
+  async activateLightConfirmation(event) {
+    event.preventDefault();
+    try {
+      const data = new FormData(event.target);
+      const response = await fetch(event.target.action, {
+        method: 'POST',
+        headers: {
+          Accept: 'application/json',
+          'X-CSRF-Token': document.querySelector("meta[name='csrf-token']")
+            ?.content,
+        },
+        body: data,
+        credentials: 'same-origin',
+      });
+
+      const outcome = await response.json();
+
+      if (response.ok) {
+        this.displaySnackbar(outcome.message);
+      } else {
+        this.displaySnackbar(outcome.error);
+      }
+    } catch (err) {
+      this.displaySnackbar(err);
+    }
+    event.target.querySelector('input[type=submit]').disabled = false;
```

Comment:
I think we need to set this `disabled` attribute to `true` at the beginning of this function - definitely seems to be possible to hammer the button and trigger a lot of the same network request 😅 

I also wonder (not really for this PR, but just a thought for future iterations) if we want to show some kind of visual indicator to a user that their update is in progress? The process is almost too smooth, so it feels like nothing is happening after I pressed submit

Reply:
aaaah actually ignore that first part as I just saw your comment about this functionality.

I do think it's a bit confusing at the moment - because I can fire of 4, 5, however many updates at once, and I don't have any indication as to which have succeeded and which have failed

Reply:
yeah I'm not sure how to counteract this ... I felt the same way as you 😓

Reply:
I'm not sure either 🤔 

Maybe the confirmation message could have the input name in it or something e.g. 'Successfully updated suggested users.'?  Maybe there's a way to highlight any inputs we fail to update by adding an `error` class or something 🤔 🤔  I'm not sure! I don't think it necessarily has to be solved in this PR though 🙂

Reply:
Ok, I will make a note of this for the near future.

Reply:
Is disabling the button from an accessibility standpoint OK in this scenario? There are more accessible patterns to handle this or is this OK in this scenario @aitchiss?

Reply:
It's a good point - the problem with disabling buttons/inputs is it effectively removes them completely from the accessibility tree (until re-enabled). It can be confusing for assistive tech users, but in the case where it happens _after_ submit it's not so bad really (since they already found the button to activate it, it's less confusing to understand why it's now disappearing, if you see what I mean).

I notice that disabling the button on submit is a default Rails behaviour, and we're having to work around it by re-enabling. I'm not sure if it's appropriate for this PR (the submit button is a shared component and maybe in some places we still want the disabled behaviour?) but we could stop Rails from disabling on submit by adding

```erb
<%= f.submit "Update Settings", class: "crayons-btn mt-4", data: {disable_with: false} %>
```

Reply:
@aitchiss Yes the Submit button is a shared component, BUT I'm only using it for the Light Confirmation flows (ie this PR 😅 )

So I think this solution you suggest will work just fine. I like that I'm not effecting the button re-enable from Stimulus.